mage_ai/server/api/downloads.py

```diff
@@ -0,0 +1,42 @@
+from mage_ai.data_preparation.models.pipeline import Pipeline
+from mage_ai.data_preparation.models.variable import VariableType
+from mage_ai.orchestration.db.models.schedules import PipelineRun
+from mage_ai.server.api.base import BaseHandler
+from tornado import gen, iostream
+
+
+class ApiDownloadHandler(BaseHandler):
+    async def get(self, pipeline_uuid, block_uuid, **kwargs):
```

Comment:
Could you add authentication to this api?

Reply:
added

Reply:
tested the API request with/without the token and confirmed that an exception is raised if no valid token was included